File: remote/src/main/java/com/kelsos/mbrc/dto/player/StatusResponse.java
package com.kelsos.mbrc.dto.player;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.kelsos.mbrc.dto.BaseResponse;

import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;
import org.apache.commons.lang.builder.ToStringBuilder;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
    "enabled",
    "success"
})
public class StatusResponse extends BaseResponse {

  @JsonProperty("enabled")
  private Boolean enabled;

  /**
   * @return The enabled
   */
  @JsonProperty("enabled")
  public Boolean getEnabled() {
    return enabled;
  }

  /**
   * @param enabled The enabled
   */
  @JsonProperty("enabled")
  public void setEnabled(Boolean enabled) {
    this.enabled = enabled;
  }

  @Override
  public String toString() {
    return ToStringBuilder.reflectionToString(this);
  }

  @Override
  public int hashCode() {
    return new HashCodeBuilder().append(enabled).toHashCode();
  }

  @Override
  public boolean equals(Object other) {
    if (other == this) {
      return true;
    }
    if (!(other instanceof StatusResponse)) {
      return false;
    }
    StatusResponse rhs = ((StatusResponse) other);
    return new EqualsBuilder().append(enabled, rhs.enabled).isEquals();
  }

}

File: mbrc-lib/src/main/kotlin/com/kelsos/mbrc/dto/player/StatusResponse.kt
package com.kelsos.mbrc.dto.player

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonPropertyOrder
import com.kelsos.mbrc.dto.BaseResponse

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder("enabled", "success")
class StatusResponse : BaseResponse() {

    /**
     * @return The enabled
     */
    /**
     * @param enabled The enabled
     */
    @JsonProperty("enabled")
    var enabled: Boolean? = null
}

